woocommerce/woocommerce-ios
# [Mobile Payments] [Reader Updates] Connect startCardReaderUpdate and UpdateViewController

Partially addresses #4059 

Changes:
- Adds dispatching of startCardReaderUpdate to CardReaderSettingsConnectedViewModel
- Allows CardReaderSettingsConnectedViewController to request that get kicked off
- Adds new flags for the viewmodel to prompt CardReaderSettingsConnectedViewController to display the UpdateViewController or success snackbar
- Adds UpdateViewController

To test:
- **IMPORTANT: To avoid updating an actual card reader, first enable `-simulate-stripe-card-reader` for the scheme before running**
- Tap on settings (gear), then In-Person Payments, then Manage card reader
- Tap connect and wait for the reader to connect
- When notified of the update, tap on update
- Wait for the update to complete
- Verify a success notice is displayed
- Note that the "update available" is no longer present

Unfinished business for a subsequent PRs:
- [ ] Displaying a failure message
- [ ] Expand unit tests to cover new viewmodel additions

Screenshots:

iPad (10.2") Portrait Dark Updating

<img src="https://user-images.githubusercontent.com/1595739/130701209-d4b4309a-b2ce-483d-90bb-4e2e16a4be0f.PNG" width=50%/>

iPad (10.2") Portrait Dark Updated

<img src="https://user-images.githubusercontent.com/1595739/130701224-e0b47724-b0fe-4df6-b22c-d9cc513f995a.PNG" width=50%/>

iPad (10.2") Landscape Dark Updating

<img src="https://user-images.githubusercontent.com/1595739/130701230-5d316d1d-c231-4259-98fe-e5d67b61f47c.PNG" width=50%/>

iPhone (12 Pro Max) Portrait Light Updating

<img src="https://user-images.githubusercontent.com/1595739/130701246-95127666-38ee-4de2-a850-e331e3af6d85.PNG" width=50%/>

iPhone (12 Pro Max) Portrait Light Updated

<img src="https://user-images.githubusercontent.com/1595739/130701265-3edc483a-f317-4dbd-b1f0-07b9c661cf9a.PNG" width=50%/>

iPad Movie

https://user-images.githubusercontent.com/1595739/130701185-c833d077-b631-4fe0-8d34-2d80c44d61c7.MP4

Update release notes:
